### PR TITLE
Rework OurShould, OurWait for add fail messages

### PR DESCRIPTION
**Pull request by estuold**
Just an idea.
Maybe it's not the best solution but as one of.

### CODE REVIEW

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
  * Created by vefimov on 18/04/2017.
  */
-public class FluentWaitCondition<T> extends FluentWait<T> {
+public class FluentWaitCondition extends FluentWait<WebDriver> {
 
     //default condition should stop execution in case of failure
     private boolean nullOnFailure = false;
 
-    public FluentWaitCondition(T input) {
+    public FluentWaitCondition(WebDriver input) {
```
Comment (vefimofff) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java]: lets rename input to driver

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java [hunk 2]
```diff
@@ -1,36 +1,35 @@
...
-            }
-        } else {
+    <T> void waitFor(Function<WebDriver, T> condition) {
+        try {
             until(condition);
+        } catch (Throwable ignoredAndContinue) {
+            if (nullOnFailure) {
+                //todo add some logging here if necessary
+            } else {
+                throw new AssertionError("Unable to perform: " + condition.toString());
```
Comment (vefimofff) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java]: let's rephrase to something like "Condition: ***'" + condition.toString() "'*** failed!"

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java [hunk 3]
```diff
@@ -24,37 +29,99 @@ public OurShould(OurWebElement element, OurSearchStrategy strategy) {
...
-        waitFor((element) -> element.getAttribute(attributeName) == null);
+        waitFor(new ExpectedCondition<Boolean>() {
+            @Override
+            public Boolean apply(WebDriver input) {
+                return element.getAttribute(attributeName) == null;
+            }
+
+            @Override
+            public String toString() {
+                return String.format("should element '%s' not have attribute '%s'", element.toString(), attributeName);
```
Comment (vefimofff) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java]: here and everywhere please replace "should element" to "element should" for better wording

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java (hunk 1 above)
Comment (estuold) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java]: Done

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java (hunk 3 above)
Comment (estuold) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java]: Done

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java (hunk 2 above)
Comment (estuold) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java]: Done

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java (hunk 1 above)
Comment (estuold) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java]: Done

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java (hunk 2 above)
Comment (estuold) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/FluentWaitCondition.java]: Done

File: src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java (hunk 3 above)
Comment (estuold) [inline src/main/java/com/wiley/autotest/selenium/elements/upgrade/v3/OurShould.java]: Done

**Comment by estuold** [comment]
Changes in #https://github.com/WileyLabs/teasy/pull/34/commits/938ef32d79c216f92af778b931b49abb42a485e5

**Pull request merged**